File: Euler OLD/primegen.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

const int ROW_SIZE=200;
const int MAX=664600;

int* strtoi(char* str){
	int found=0;
	int* ret = malloc(sizeof(int)*(5));
	int i=0;
	while(found<5){
		char temp[6];
		int j=0;
		while(str[i]!=' '){
			temp[j]=str[i];
			i++;
			j++;
		}
		ret[found]=atoi(temp);
		found++;
	}
	return ret;
}

int main(){
	/*FILE *f = fopen("cprimes.csv", "r");
	int p=2;
	int [ROW_SIZE] lst;

	//get the first ROW_SIZE primes
	int i=0;
	while(i<ROW_SIZE){
		if(p==2){
			lst[i]=p;
			i++;
			p++;
		}
		else{
			int j=0;
			int good=1;
			while(j<i && j<=(int)ceil(sqrt((double)p)) && good==1 ){
				good = (p%lst[j]==0) ? 0 : 1;
				j++;
			}
			if(good==1){
				lst[i]=p;
				i++;
			}
			p+=2;
		}
	}

	int largest = lst[ROW_SIZE-1];
	//Write first row to file
	for(i=0; i<ROW_SIZE; i++){
		fprintf(f, "%d ", lst[i]);
	}
	fprintf(f, "\n");
	fclose(f);

	//hard part
	int rowsWritten=1;
	p=largest+2;
	while(rowsWritten<MAX/ROW_SIZE){
		i=0;
		while(i<ROW_SIZE){
			int dead=0;
			f = fopen("cprimes.csv", "r");
			int j=0;
			while(j<rowsWritten){

			}
		}
	}*/
	char* teststr = "1 2 3 4 5";
	int* formattest = strtoi(teststr);
	int i;
	for(i=0;i<5;i++)
		printf("%d ", formattest[i]);
	return 0;
}

File: Euler OLD/primegen.py
#
#	Creates a csv for first million primes
#	100 cols. Check primality by iterating through current list of primes
#	Start by generating the first 150 with a naive sieve.

import math, csv

primes = open("primes.csv", "r+")
writer = csv.writer(primes, delimiter = ",")

ROWSIZE=200
MAX=664600

#	First chunk of primes
p=2
lst = []
while len(lst)<ROWSIZE:
	if p==2:
		lst.append(p)
	else:
		i=0
		good = True
		while i<len(lst) and lst[i]<=math.ceil(math.sqrt(p)) and good==True:
			good = False if p%lst[i]==0 else True
			i=i+1
		if good==True:
			lst.append(p)
	p = 3 if p==2 else p+2

largest = lst[ROWSIZE-1]
writer.writerow(lst)
primes.close()

# moar primes
written=1
p=largest+2
while written<MAX/ROWSIZE:
	nuRow=[]
	while len(nuRow)<ROWSIZE:
		dead=False
		primes = open("primes.csv", "r+")
		writer = csv.writer(primes, delimiter = ",")
		reader = csv.reader(primes, delimiter = ",")
		for row in reader:
			i=0
			good=True
			while i<ROWSIZE and int(row[i])<=int(math.ceil(math.sqrt(p))) and good==True:
				good = False if p%int(row[i])==0 else True
				i=i+1
			if good==False:
				dead=True
				break
		if dead==False:
			nuRow.append(p)
		p=p+2
	writer.writerow(nuRow)
	if(written%5==0):
		print written*200
	written = written+1
primes.close()